crates/transaction-pool/src/traits.rs

```diff
@@ -138,7 +147,10 @@ pub trait TransactionPool: Send + Sync + Clone {
     fn new_pending_pool_transactions_listener(
         &self,
     ) -> NewSubpoolTransactionStream<Self::Transaction> {
-        NewSubpoolTransactionStream::new(self.new_transactions_listener(), SubPool::Pending)
+        NewSubpoolTransactionStream::new(
+            self.new_transactions_listener_for(PendingTransactionListenerKind::PropagateOnly),
+            SubPool::Pending,
+        )
```

Comment:
I didn't do the same refactor for `SubPool::BaseFee` or `SubPool::Queued` (right below this function) because I didn't see them used anywhere.

Reply:
this is reasonable, and we don't really have a usecase for this anyway